ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/ForeignKeyTableTableJoinBuilder.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.streams;
+
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.execution.plan.ForeignKeyTableTableJoin;
+import io.confluent.ksql.execution.plan.KTableHolder;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import org.apache.kafka.streams.kstream.KTable;
+
+public final class ForeignKeyTableTableJoinBuilder {
+
+  private ForeignKeyTableTableJoinBuilder() {
+  }
+
+  public static <KLeftT, KRightT> KTableHolder<KLeftT> build(
+      final KTableHolder<KLeftT> left,
+      final KTableHolder<KRightT> right,
+      final ForeignKeyTableTableJoin<KLeftT, KRightT> join
+  ) {
```

Comment:
What if we add info to the current `TableTableJoin` where we could check if the join is on a foreign key? Maybe we can reuse the `TableTableJoinBuilder`, and just change the impl. to call a different builder for foregign keys and another that has the current code.

Reply:
We could, but it seems cleaner to have a dedicated node for FK-joins in the physical plan, as mentioned by Victory above.